drivers/modem/modem_receiver.c

```diff
@@ -119,7 +115,9 @@ int mdm_receiver_recv(struct mdm_receiver_context *ctx,
 		return -EINVAL;
 	}
 
-	return k_pipe_get(&ctx->uart_pipe, buf, size, bytes_read, 1, K_NO_WAIT);
+	*bytes_read = ring_buf_get(&ctx->rx_rb, buf, size);
+
+	return (*bytes_read == 0 ? -EIO : 0);
```

Comment:
Let's put this back to just returning 0 here (sorry for the noise).  I'll put up the PR for Wistron modem fix in a bit.   bytes_read == 0 isn't really an error.

Reply:
Well... let it be so